src/base-widget.ts

```diff
@@ -1,65 +1,42 @@
 import { Input, OnDestroy, OnInit } from '@angular/core';
 import { isPlatformBrowser } from '@angular/common';
 import { bem, noop } from './utils';
-
-export class Widget {
-  public init: (
-    params: {
-      templatesConfig: object;
-      state: object;
-      results: object[];
-      createURL: (value: any) => string;
-      instantSearchInstance: object;
-    }
-  ) => void;
-  public getConfiguration?: () => object;
-  public render: (
-    params: {
-      templatesConfig: object;
-      state: object;
-      results: {}[];
-      createURL: (value: any) => string;
-      instantSearchInstance: object;
-    }
-  ) => void;
-  public dispose: (
-    params: {
-      helper: object;
-      state: object;
-    }
-  ) => object | void;
-}
+import { NgAisInstantSearch } from './instantsearch/instantsearch';
+import { Widget } from 'instantsearch.js/es/types';
+export { Widget };
 
 export type Connector = (
   renderFn: (state: object, isFirstRendering: boolean) => void,
   unmountFn: () => void
 ) => (widgetOptions?: object) => Widget;
 
-export class BaseWidget implements OnInit, OnDestroy {
-  public instantSearchParent: any;
-
+export abstract class BaseWidget implements OnInit, OnDestroy {
```

Comment:
note well that this is now an abstract class, making it error when someone forgets to implement `instantSearchParent` (before that silently failed)